types/sumo-logger/index.d.ts

```diff
@@ -0,0 +1,128 @@
+         * `window.location` value.
+         */
+        clientUrl?: string;
+
+        /**
+         * Setting `sendErrors` to `true` will send all the unhandled errors to Sumo Logic
+         * with the error message, URL, line number, and column number.
+         * This attribute plays well with any other `window.onerror` functions that
+         * have been defined.
+         */
+        sendErrors?: boolean;
+
+        /** To identify specific user sessions, set a value for this field. */
+        sessionKey?: string;
+
+        /**
+         * This value identifies the host from which the log is being sent.
+         */
+        hostName?: string;
+
+        /**
+         * This value sets the Source Category for the logged message.
+         */
+        sourceCategory?: string;
+
+        /**
+         * This value sets the Source Name for the logged message.
+         */
+        sourceName?: string;
+    }
+
+    interface PerMessageOptions {
+        /**
+         * Defaults to `new Date()` called when processing the log call.
+         * Use this when the event being logged occurred
+         * at a different time than when the log was sent.
+         */
+        timestamp: Date;
+
+        /** Override a session key set in the `config` call. */
+        sessionKey: string;
+
+        /** Override client URL set in the config call. (Node version only) */
+        url: string;
+    }
+}
+
+/**
+ * You must have an HTTP source created in your Sumo Logic account to use this SDK.
+ * To create one, log into Sumo Logic, go to the Collectors page and either create
+ * a new Hosted Collector or add a new HTTP source to an existing Hosted Collector.
+ */
+declare class SumoLogger {
+    constructor(options: SumoLogger.SumoLoggerOptions);
+
+    /**
+     * **(Vanilla JS lib only)**
+     * Set the configuration for sending logs. Options are listed in the next section.
+     * In the Node.js module, configuration options are sent when instantiating the object.
+     */
+    config?(options: SumoLogger.SumoLoggerOptions): void;
+
+    /**
+     * Set a log message to be sent.
+     * All logs are sent as JSON objects.
+     * If you call `log()` with just a string, the string is included as a field called msg.
+     * If you call the function with a JSON object, each field in the object is included as a separate field.
+     * Fields called `sessionId`, `url`, and `timestamp` are sent in both cases.
+     */
+    log(message: string, options?: SumoLogger.PerMessageOptions): void;
+
+    /**
+     * Set a log message to be sent.
+     * All logs are sent as JSON objects.
+     * If you call `log()` with just a string, the string is included as a field called msg.
+     * If you call the function with a JSON object, each field in the object is included as a separate field.
+     * Fields called `sessionId`, `url`, and `timestamp` are sent in both cases.
+     */
+    log<T extends { }>(event: Partial<SumoLogger.PerMessageOptions> & T): void;
```

Comment:
Yes, I started with `object` but `npm run lint` failed because `object` is TS 2.2+. I'd like to make it clear that we need a plain JS object.